### DISCUSSION TITLE
Flashing ESP-02S module with Tasmota

### DISCUSSION THREAD

**Discussion by whabby** [arendst/Tasmota#17910] [Q&A]
Hi everybody
I tried to flash Tasmota light on ESP-02S modules. This module ist ESP 8286.
First I used the web installer (https://tasmota.github.io/install/). It was flashing without error, but after flashing and reboot, there was no Wifi-Accesspoint available. Tried the second module on the same way, everything went well and the module works perfect. Tried to flash the next 3 modules, same success as module #1, no acces point.
Tried to flash the modules again with Tasmotizer -> MD5 of file does not match data in flash!
Tried to flash the modules with ESP-flasher after reaching 100% ->  Unexpected error: Error while writing flash: Serial data stream stopped: Possible serial noise or corruption.

Has anyone an idea what could be wrong? What could be the reason that one module works fine and four modules don't work? I know, it's always possible that a module don't work, but 4 of 5 is special.

**Comment by barbudor** [comment]
What is your USB serial adpater ?
Flashing process requires quite some current and many adpaters doesn't have a proper regulator to feed the ESP during the flashing
Avoid any device that feed the 3V3 from the serial adapter chpi. It must have a dedicated 3V3 regulator

You probably just have been lucky with module #2

**Reply by whabby** [reply]
I use https://de.aliexpress.com/item/33043136429.html?spm=a2g0o.order_list.order_list_main.4.21ef5c5f55HwG4&gatewayAdapt=glo2deu as USB serial adapter. Until now, i could flash every module with that - exept now the ESP-02S. I will order a other adapter and try to flash again. Thanks for the idea!

**Reply by barbudor** [reply]
The worst available
Tasmota team generally recommend this one : https://de.aliexpress.com/w/wholesale-golden-ch340.html?catId=0&initiative_id=SB_20230207131705&origin=y&SearchText=golden+ch340&spm=a2g0o.detail.1000002.0

But my personnal preference goes for this one : https://de.aliexpress.com/item/1005002064355636.html
which is dedicated for ESP. It has logic to control GPIO0 and RST which makes flashing very simple (even if not using ESP01)

See the big 3V3 regulator at the arrow: you can clearly read AMS1117 which is :+1: 

![image](https://user-images.githubusercontent.com/2996042/217368175-5d14e376-59f8-4274-a4eb-a363f39258f3.png)

**Reply by whabby** [reply]
Ordered your personal preference just now. Will take about 2 weeks until I can test it. I will report ...

**Comment by whabby** [comment]
Hi Barbudor
Recieved your personal preference today and tested immediately. Same result as with the module before. Any other idea ?

Regards 

 

Von: Barbudor ***@***.***> 
Gesendet: Dienstag, 7. Februar 2023 22:21
An: arendst/Tasmota ***@***.***>
Cc: whabby ***@***.***>; Author ***@***.***>
Betreff: Re: [arendst/Tasmota] Flashing ESP-02S module with Tasmota (Discussion #17910)

 

The worst available
Tasmota team generally recommend this one : https://de.aliexpress.com/w/wholesale-golden-ch340.html?catId=0 <https://de.aliexpress.com/w/wholesale-golden-ch340.html?catId=0&initiative_id=SB_20230207131705&origin=y&SearchText=golden+ch340&spm=a2g0o.detail.1000002.0> &initiative_id=SB_20230207131705&origin=y&SearchText=golden+ch340&spm=a2g0o.detail.1000002.0

But my personnal preference goes for this one : https://de.aliexpress.com/item/1005002064355636.html

See the big 3V3 regulator at the arrow: you can clearly read AMS1117 which is 👍

 <https://user-images.githubusercontent.com/2996042/217368175-5d14e376-59f8-4274-a4eb-a363f39258f3.png>

**Reply by barbudor** [reply]
Hi @whabby 

How are you connecting to the module ?
Are you sure of the contacts ?
If possible try to do a light solder at least on the module side

Beside that or a whole pack of bad luck with 4 defective modules out of 5, I'm starting to lack ideas.

**Comment by habo20** [comment]
@whabby 
I have the same problem, after flashing a new ESP-02S module without error, I get no access point.
Do you have a solution in the meantime, or were your modules simply defective?

**Reply by habo20** [reply]
I have found the solution, the GPIO0-GND connection must be disconnected, then the software starts as usual with an access point

**Comment by Saint6822** [comment]
I know this is an old thread, but have you verified what it takes for these modules to boot after programming any pins need to be held low?

**Reply by sfromis** [reply]
This page discusses ESP8266 gpios, including what pins will inhibit boot if low or high.
https://randomnerdtutorials.com/esp8266-pinout-reference-gpios/
(See section "Best Pins to Use")

Logically, doing the inverse should allow the flashed firmware to boot. IDK if the ESP-02S already does some of this.

**Comment by dummbold** [comment]
Hi all,

I want to joyn this Thread rather than opening a new one.

Some weeks ago I ordered a couple of these Chips from eBay (China) for some little smarthome-projects (eg. upgrading a small LED spotlight with pid and pv-loading to become smart). The vendor wrote that they are Tasmota pre-flashed, but they were not. It's Tuya firmware on board. So I tried to tasmotize them (this is all completely new to me...). As I am a Mac-User I tried first with the Mac-Tasmotizer, but this App was unable to establish a connection. Then I tried the webpage (most hated Chrome browser). But this was also unsuccessful. It says:

<img width="404" alt="screen" src="https://github.com/user-attachments/assets/af5376e1-98e9-4028-8e68-096456d89c34" />

I use a serial Adapter based on the CH340G (from AZ-Delivery), and I installed the drivers on the Mac. But the Adapter has no "Reset-Button" or anything like that. I can also see the AP as soon as the Chip receives Power.

Is it possible that the whole thing is just not running on an M4-Mini with actual Sequoia?

**Reply by sfromis** [reply]
You need the correct wiring, and you told nothing about what you did for that. These pin are the minimum, and gpio 0 is the "boot button" to be grounded before/while powering up. 
![image](https://github.com/user-attachments/assets/3d5ef6a1-e0c5-4239-b68f-62334267b011)

**Reply by dummbold** [reply]
Thanks very much for this information. Unfortunately there is only little documentation for the ESP-02S. So I wired only VCC, GND and RXD/TXD (inverted). The Pins shown here do not exist on the ESP-02S!

But ... I was lucky! This guy did it and documented it:

https://alexwtz.ch/blog/portfolio/energy-monitoring-with-cheap-smart-plug/

I just have to ground IO0. So I will try this out.
Sometimes it's worth doing just more google search with different wordings.

**Reply by sfromis** [reply]
Some modules do include part of the needed wiring internally.
Pinout: https://templates.blakadder.com/ESP-02S.html

**Comment by dummbold** [comment]
Okay,
and here's the very final **How-To (tasmotize ESP-02S)** for all those dumb beginners (like me):

1. Get a USB to serial adapter (like mentioned above). I have one based on the GH340 which ist recommended here. Vendor: AZ-Delivery
2.  Set the jumpers to 3V3 (essential!)
3. Wire the following (Adapter to Chip):
         3V3 -> 3V3
         GND -> GND
         GND -> IO0 (the little Pin is 1cm above the GND Pin)
         RXD -> TX (_send_ and _receive_ must be crossed!!!)
         TXD -> RX
4. I tried the Tasmotizer (Mac-Version) on a M4, but it was only possible to erase the old Firmware (Tuya), not to write Tasmota, so I would recommend to install Chrome and use the Online-Version. Due to privacy issues it runs on a separate user on my Mac without any access to NAS or Cloud - just in case ...
5. After successful installation plug-out the Adapter from USB.
6. Desolder GND from IO0 (essential!)
7. Plugin the adapter again
8. Now you should see the AP from the chip named "tasmota-blablablanumbers"
9. Access the AP and wait a few moments - UI starts itself
10. Establish connection to your network
11. Be happy - you made it! 🫡  🤗

Now you should have a tiny little and cheap Chip for all the small smarthome projects that come into your mind ...

Edit: it should also be possible to replace or reflash identical or similar Tuya-Chips inside those badass spyware devices.
